### DISCUSSION TITLE
Is there a way to disable the creation of a YAML header while converting from ipynb to qmd?

### DISCUSSION THREAD

**Discussion by bernardo-cruz** [quarto-dev/quarto-cli#5228] [Q&A] [answered]
I am working on a report consists of collection of separate `ipynb` files. However, once the `ipynb` document in concluded, I convert the file using cli, `quarto convert document.ipynb`. The problem is, that this command adds a YAML header to the `qmd` file and adds the first section of the text (#) in the YAML header as title - see example below. 

```
---
title: First Heading of ipynb
jupyter: python3
---
```

Is there a way to disable the creation of a YAML header while converting from `ipynb` to `qmd`?

**Comment by mcanouil** [comment]
A quarto document requires a YAML header within (or as a global configuration file `_quarto.yml`).
Could you elaborate on what's the goal of `ipynb` -> `qmd` here? _i.e._, Quarto can render Jupyter notebook.

**Reply by bernardo-cruz** [reply]
thx for your answer :)

I have a main file (`.qmd`). with the yaml header. The main file includes the converted `ipynb` using the command `{{< include converted_from_ipynb.qmd >}}`. I have tried to use the `{{< include document.ipynb>}}`, but I didn't work. As far as I have seen in the documentation, one cannot use `include` in `.qmd` from a `ipynb` source. 

For that reason I create each section using `ipynb`, which is easier to debug etc. and once finished, I convert it to `.qmd` and add the `qmd` to the main file using `include` - see example below:

```
---
YAML content
---

# Introduction
{{< include converted_from_ipynb.qmd >}}

# Section One
{{< include section_one_converted_from_ipynb.qmd >}}

# Section Two
{{< include section_two_converted_from_ipynb.qmd >}}

...
```

**Reply by mcanouil** [reply, marked as answer]
Yes, it is stated in the documentation that the include shortcode does not work for `ipynb` files, see <https://quarto.org/docs/authoring/includes.html#computations>.

This being said, maybe you want the **embed** feature from v1.3, see <https://quarto.org/docs/authoring/notebook-embed.html>.

If this does not fit your needs, I would suggest to fully embrace Quarto, and directly write your "components" as Quarto documents.

**Reply by bernardo-cruz** [reply]
Oh I see, I did not know about the `embed` feature. Thx for the hint!

**Reply by bernardo-cruz** [reply]
BTW: I just found out that quarto version v1.2.475 does not show this behavior. When converting from `ipynb` to `qmd` , no field `title` is created in the YAML header.

**Reply by mcanouil** [reply]
@dragonstyle @cscheid do you have any insights on this change between 1.2 and 1.3?

**Reply by cscheid** [reply]
That change is by design: H1 headings are meant to be "this is the overall title of the document", and are now converted as such when going from ipynb to qmd. That was a 1.2 bug.

The problem here, I think, is that @bernardo-cruz is using the converted file as a subdocument, but that .ipynb has elements that are interpreted as "top-level", such as the title. (This is fundamentally the same problem that happens if you were to call `{{< include foo.qmd >}}` and `foo.qmd` had metadata: it will be interpreted as the metadata of the parent document, by design. As we explain in the `include` documentation, `include` behaves literally like copying and pasting.)

If you intend to use the result of `quarto convert`ing a ipynb file as an include, then it shouldn't have markup that is interpreted as "top-level" information: use `##` instead of `#`. If you're using `##` and we're still putting that in the YAML, that's a bug we need to fix.

**Reply by cderv** [reply]
Curious if the pandoc options to shift header level ([`--shift-heading-level-by`](https://pandoc.org/MANUAL.html#reader-options)) could be used when converting such document having higher section to have lower section in .qmd for using in includes.

But we don't use Pandoc for `quarto convert` right @cscheid ?

**Reply by cscheid** [reply]
@cderv No, we use https://github.com/quarto-dev/quarto-cli/blob/main/src/command/convert/jupyter.ts#L39-L129

**Reply by cderv** [reply]
Thanks for pointing me to the code. Built in feature then - good to know !

**Comment by mcanouil** [comment]
FYI, #5363 for a related issue.
